### PR TITLE
Use a highlight modifier for intra doc links

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

File: crates/ide/src/syntax_highlighting/inject.rs [hunk 1]
```diff
@@ -243,10 +246,10 @@ pub(super) fn doc_comment(hl: &mut Highlights, sema: &Semantics<RootDatabase>, n
         }
     }
 
-    for range in intra_doc_links {
+    for (def, range) in intra_doc_links {
         hl.add(HlRange {
             range,
-            highlight: HlTag::IntraDocLink | HlMod::Documentation,
+            highlight: module_def_to_hl_tag(def) | HlMod::Documentation | HlMod::IntraDocLink,
```
Comment (Veykril) [inline crates/ide/src/syntax_highlighting/inject.rs:252]: Should this also get the `Injected` Modifier? I'm unsure what exactly injection refers to, is it only source code highlighting or more general?

**Comment by Veykril** [comment]
Called the modifier things `IntraDocLink` and the like but maybe dropping the `Intra` to be more general is better for the future?

r? @matklad

**Comment by lnicola** [comment]
changelog feature add `intra_doc_link` semantic highlighting modifier

**Pull request merged**